### PR TITLE
Remove legacy DataFrame implementation

### PR DESCRIPTION
**Pull request by phofl**
Closes #xxxx

- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

sibling of https://github.com/dask/dask/pull/11606

### CODE REVIEW

File: continuous_integration/environment-3.11.yaml [hunk 1]
```diff
@@ -49,8 +49,8 @@ dependencies:
   # Temporary fix for https://github.com/jupyterlab/jupyterlab/issues/17012
   - httpx<0.28.0
   - pip:
-      - git+https://github.com/dask/dask
-      - git+https://github.com/dask/dask-expr
+      - git+https://github.com/phofl/dask@remove-legacy-implementation
+      - git+https://github.com/phofl/dask-expr@legacy-conversion
```
Comment (phofl) [inline continuous_integration/environment-3.11.yaml:53]:
```suggestion
      - git+https://github.com/dask/dask-expr
```

File: continuous_integration/environment-3.12.yaml [hunk 2]
```diff
@@ -49,8 +49,8 @@ dependencies:
   # Temporary fix for https://github.com/jupyterlab/jupyterlab/issues/17012
   - httpx<0.28.0
   - pip:
-      - git+https://github.com/dask/dask
-      - git+https://github.com/dask/dask-expr
+      - git+https://github.com/phofl/dask@remove-legacy-implementation
+      - git+https://github.com/phofl/dask-expr@legacy-conversion
```
Comment (phofl) [inline continuous_integration/environment-3.12.yaml:53]:
```suggestion
      - git+https://github.com/dask/dask-expr
```

File: continuous_integration/environment-3.13.yaml [hunk 3]
```diff
@@ -48,8 +48,8 @@ dependencies:
   # Temporary fix for https://github.com/jupyterlab/jupyterlab/issues/17012
   - httpx<0.28.0
   - pip:
-      - git+https://github.com/dask/dask
-      - git+https://github.com/dask/dask-expr
+      - git+https://github.com/phofl/dask@remove-legacy-implementation
+      - git+https://github.com/phofl/dask-expr@legacy-conversion
```
Comment (phofl) [inline continuous_integration/environment-3.13.yaml:52]:
```suggestion
      - git+https://github.com/dask/dask-expr
```

File: continuous_integration/environment-3.10.yaml [hunk 4]
```diff
@@ -55,8 +55,8 @@ dependencies:
   # Temporary fix for https://github.com/jupyterlab/jupyterlab/issues/17012
   - httpx<0.28.0
   - pip:
-      - git+https://github.com/dask/dask
-      - git+https://github.com/dask/dask-expr
+      - git+https://github.com/phofl/dask@remove-legacy-implementation
+      - git+https://github.com/phofl/dask-expr@legacy-conversion
```
Comment (phofl) [inline continuous_integration/environment-3.10.yaml:59]:
```suggestion
      - git+https://github.com/dask/dask-expr
```

File: continuous_integration/environment-mindeps.yaml [hunk 5]
```diff
@@ -23,8 +23,8 @@ dependencies:
   # Distributed depends on the latest version of Dask
   - pip
   - pip:
-    - git+https://github.com/dask/dask
-  # test dependencies
+      - git+https://github.com/phofl/dask@remove-legacy-implementation
+    # test dependencies
```
Comment (phofl) [inline continuous_integration/environment-mindeps.yaml:27]:
```suggestion
  # test dependencies
```

File: continuous_integration/environment-3.10.yaml [hunk 6]
```diff
@@ -55,7 +55,7 @@ dependencies:
   # Temporary fix for https://github.com/jupyterlab/jupyterlab/issues/17012
   - httpx<0.28.0
   - pip:
-      - git+https://github.com/dask/dask
+      - git+https://github.com/phofl/dask@remove-legacy-implementation
```
Comment (hendrikmakait) [inline continuous_integration/environment-3.10.yaml:58]: TODO: Revert these changes

File: continuous_integration/environment-3.11.yaml [hunk 7]
```diff
@@ -49,7 +49,7 @@ dependencies:
   # Temporary fix for https://github.com/jupyterlab/jupyterlab/issues/17012
   - httpx<0.28.0
   - pip:
-      - git+https://github.com/dask/dask
+      - git+https://github.com/phofl/dask@remove-legacy-implementation
```
Comment (hendrikmakait) [inline continuous_integration/environment-3.11.yaml:52]: TODO: Revert these changes

File: continuous_integration/environment-3.12.yaml [hunk 8]
```diff
@@ -49,7 +49,7 @@ dependencies:
   # Temporary fix for https://github.com/jupyterlab/jupyterlab/issues/17012
   - httpx<0.28.0
   - pip:
-      - git+https://github.com/dask/dask
+      - git+https://github.com/phofl/dask@remove-legacy-implementation
```
Comment (hendrikmakait) [inline continuous_integration/environment-3.12.yaml:52]: TODO: Revert these changes

File: continuous_integration/environment-3.13.yaml [hunk 9]
```diff
@@ -48,7 +48,7 @@ dependencies:
   # Temporary fix for https://github.com/jupyterlab/jupyterlab/issues/17012
   - httpx<0.28.0
   - pip:
-      - git+https://github.com/dask/dask
+      - git+https://github.com/phofl/dask@remove-legacy-implementation
```
Comment (hendrikmakait) [inline continuous_integration/environment-3.13.yaml:51]: TODO: Revert these changes

File: continuous_integration/environment-mindeps.yaml [hunk 10]
```diff
@@ -23,7 +23,7 @@ dependencies:
   # Distributed depends on the latest version of Dask
   - pip
   - pip:
-    - git+https://github.com/dask/dask
+      - git+https://github.com/phofl/dask@remove-legacy-implementation
```
Comment (hendrikmakait) [inline continuous_integration/environment-mindeps.yaml:26]: TODO: Revert these changes

**Review by hendrikmakait — APPROVED** [review]
Thanks, @phofl. LGM as long as CI is green.

File: continuous_integration/environment-mindeps.yaml (hunk 10 above)
Comment (phofl) [reply continuous_integration/environment-mindeps.yaml:26]:
```suggestion
      - git+https://github.com/dask/dask
```

File: continuous_integration/environment-3.13.yaml (hunk 9 above)
Comment (phofl) [reply continuous_integration/environment-3.13.yaml:51]:
```suggestion
      - git+https://github.com/dask/dask
```

File: continuous_integration/environment-3.10.yaml (hunk 6 above)
Comment (phofl) [reply continuous_integration/environment-3.10.yaml:58]:
```suggestion
      - git+https://github.com/dask/dask
```

File: continuous_integration/environment-3.12.yaml (hunk 8 above)
Comment (phofl) [reply continuous_integration/environment-3.12.yaml:52]:
```suggestion
      - git+https://github.com/dask/dask
```

File: continuous_integration/environment-3.11.yaml (hunk 7 above)
Comment (phofl) [reply continuous_integration/environment-3.11.yaml:52]:
```suggestion
      - git+https://github.com/dask/dask
```

File: continuous_integration/environment-mindeps.yaml [hunk 11]
```diff
@@ -23,7 +23,7 @@ dependencies:
   # Distributed depends on the latest version of Dask
   - pip
   - pip:
-    - git+https://github.com/dask/dask
+      - git+https://github.com/dask/dask
```
Comment (phofl) [inline continuous_integration/environment-mindeps.yaml:26]:
```suggestion
    - git+https://github.com/dask/dask
```
